frontend/app/page.tsx

```diff
@@ -101,45 +89,80 @@ export default function Home() {
+      { label: "LYX Burned:", value: `${divideBigIntTokenBalance(lyxBurned, 18).toString()} LYX` },
+    ])
+  };
 
   useEffect(() => {
-    const fetchStats = async () => {
-      const iterations = await burntPixArchives.getTotalIterations();
-      const contributors = await burntPixArchives.getTotalContributors();
-      const totalSupply = await burntPixArchives.totalSupply();
-      const supplyCap = await burntPixArchives.tokenSupplyCap();
-      const lyxBurned = await burntPixArchives.getTotalFeesBurnt();
-
-      setIterations(iterations.toString());
-      setContributors(contributors.toString());
-      setArchiveMints(`${totalSupply.toString()} / ${supplyCap.toString()}`);
-      setLyxBurned(`${divideBigIntTokenBalance(lyxBurned, 18).toString()} LYX`);
+    const fetchImmutableStats = async () => {
+      console.log("fetching immutable data");
+      const [burntPicId, winnerIterations, supplyCap] = await Promise.all([burntPixArchives.burntPicId(), burntPixArchives.winnerIters(), burntPixArchives.tokenSupplyCap()]);
+      setBurntPicId(burntPicId);
+      setWinnerIterations(new Intl.NumberFormat('en-US').format(Number(winnerIterations)));
+      setSupplyCap(Number(supplyCap));
     };
 
-    fetchStats();
+    fetchImmutableStats();
   }, []);
+
+  useEffect(() => {
+    supplyCap && fetchCollectionStats();
+  }, [supplyCap]);
+
+  const externalFetchArchives: IFetchArchives = useCallback(async (startFrom, amount, setArchives, setLoadedIndices, ownerProfiles, setOwnerProfiles) => {
+    const archiveCount = await burntPixArchives.archiveCount();
+    const count = Number(archiveCount);
+    const end = Math.min(startFrom + amount, count);
+    const newArchives: IArchive[] = [];
+
+    for (let i = startFrom; i < end; i++) {
+      const id = numberToBytes32(i + 1);
+      const archive = await burntPixArchives.burntArchives(id);
+      let isMinted = false;
+      let ownerAddress = archive.creator;
+
+      try {
+        ownerAddress = await burntPixArchives.tokenOwnerOf(id);
+        isMinted = true;
+      } catch (error) {
+        console.log(`archive ${id} not minted yet or error fetching owner`);
+      }
+
+      let ownerProfile = ownerProfiles[ownerAddress];
+      if (!ownerProfile) { // Check if the profile is not already fetched
+        ownerProfile = await getProfileData(ownerAddress, networkConfig.rpcUrl);
+        setOwnerProfiles(prevProfiles => ({...prevProfiles, [ownerAddress]: ownerProfile}));
+      }
+      let ownerAvatar;
+
+      if (ownerProfile.profileImage && ownerProfile.profileImage.length) {
+        ownerAvatar = `${constants.IPFS_GATEWAY}/${ownerProfile.profileImage[0].url.replace("ipfs://", "")}`;
+      }
+
+      newArchives.push({
+        id,
+        image: hexToText(archive.image),
+        ownerAddress,
+        ownerName: ownerProfile.name,
+        ownerAvatar,
+        isMinted,
+      });
+    }
```

Comment:
we should maybe this into the archives component as it's a lot of logic on the home page

was about to do the same for live stats but didn't want to conflict here

Reply:
yep. will work on refactor next stage as there's a lot here as is @democrazyfi